QueueCustomImplementation/Queue.cs

```diff
@@ -0,0 +1,92 @@
+﻿using QueueCustomImplementation.Interfaces;
+
+namespace QueueCustomImplementation
+{
+    public class Queue<T> : IGenericQueue<T>
+    {
+        private int _elementCount;
+        private int _maxCapacity;
+        private T[] _elements;
+
+        public Queue(int queueElementsCapacity)
+        {
+            _maxCapacity = queueElementsCapacity;
+            _elements = new T[_maxCapacity];
+            _elementCount = 0;
+        }
+
+        public T Dequeue()
+        {
+            var firstItem = _elements[0];
+
+            Drop();
+
+            return firstItem;
+        }
+
```

Comment:
Drop has to be private not to provide the possibility to call it directly

Reply:
Fixed